Fix flaky playwright test for timezone toggle

The playwright test `TestCompleteSessionOnboardingFlow.test_complete_session_onboarding_flow` is failing intermittently in CI when we test the timezone toggle.

It can be run explicitly with `uv run pytest -m playwright -k TestCompleteSessionOnboardingFlow --headed`. Dropping a `page.pause()` before the failure will open up the playwright debugger and allow things to be inspected.

Failure from failed action: https://github.com/djangonaut-space/wagtail-indymeet/actions/runs/20149536927/job/57838798605


```
=================================== FAILURES ===================================
_ TestCompleteSessionOnboardingFlow.test_complete_session_onboarding_flow[chromium] _

self = <tests.test_complete_session_onboarding_flow.TestCompleteSessionOnboardingFlow object at 0x7fa5278e0550>
mock_captcha = <MagicMock name='validate' id='140347282333904'>
page = <Page url='http://localhost:47001/sessions/test-session-2024/teams/282/'>
db = None, pause_at_end = False

    @patch("django_recaptcha.fields.ReCaptchaField.validate", return_value=True)
    def test_complete_session_onboarding_flow(
        self, mock_captcha, page: Page, db, pause_at_end
    ):
        """
        Complete end-to-end test for the session flow.
    
        Tests the entire workflow from user signup to team member access.
    
        Use --pause-at-end flag to pause at the end for manual exploration:
            uv run pytest -m playwright -k 'test_complete_session_onboarding_flow' \
                --headed --pause-at-end
        """
        # ======================================================================
        # PHASE 1: Navigator and Captain onboard
        # ======================================================================
    
        navigator = self._navigator_signs_up_and_sets_availability(page)
        captain = self._captain_signs_up_and_sets_availability(page)
    
        # ======================================================================
        # PHASE 2: Admin sets up session structure
        # ======================================================================
    
        superuser, session = self._admin_creates_session(page)
        survey = self._admin_creates_application_survey(page, session)
        project, team = self._admin_creates_project_and_team(page, session)
        self._admin_adds_navigator_and_captain_to_team(
            page, session, team, navigator, captain
        )
        self._admin_adds_themselves_as_organizer(page, superuser, session)
    
        # ======================================================================
        # PHASE 3: Djangonauts apply to session
        # ======================================================================
    
        djangonaut_a = self._djangonaut_a_applies_to_session(page, survey)
        djangonaut_b = self._djangonaut_b_applies_to_session(page, survey)
        djangonaut_c = self._djangonaut_c_applies_to_session(page, survey)
    
        # ======================================================================
        # PHASE 4: Admin forms teams and sends notifications
        # ======================================================================
    
        self._admin_assigns_djangonaut_a_to_team(
            page, superuser, session, team, project, djangonaut_a
        )
        self._admin_adds_djangonaut_b_to_waitlist(page, djangonaut_b)
        self._admin_sends_session_result_notifications(page, session)
    
        self._djangonaut_a_accepts_invitation(page, session)
    
        self._admin_promotes_djangonaut_b_from_waitlist_to_team(
            page, superuser, session, team, project, djangonaut_b
        )
        self._admin_sends_acceptance_email_to_djangonaut_b(page, session, djangonaut_b)
    
        self._djangonaut_b_accepts_invitation(page, session)
    
        self._admin_send_team_welcome_emails(page, superuser, session)
    
        # ======================================================================
        # PHASE 5: Team members verify access
        # ======================================================================
    
        self._verify_all_team_members_can_access_team(page, session, team, False)
    
        # ======================================================================
        # PHASE 6: Test timezone toggle JavaScript functionality
        # ======================================================================
    
        # Login as navigator (who has availability and overlaps with team members)
        page.goto(reverse("login"))
        page.locator("#id_username").fill("navigator1")
        page.locator("#id_password").fill("testpass123")
        page.get_by_role("button", name="Login").click()
        page.wait_for_load_state("networkidle")
    
>       self._test_timezone_toggle_functionality(page, session, team)

tests/test_complete_session_onboarding_flow.py:1231: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_complete_session_onboarding_flow.TestCompleteSessionOnboardingFlow object at 0x7fa5278e0550>
page = <Page url='http://localhost:47001/sessions/test-session-2024/teams/282/'>
session = <Session: Test Session 2024>, team = <Team: Team Alpha - Django>

    def _test_timezone_toggle_functionality(
        self, page: Page, session: Session, team: Team
    ):
        """
        Test the timezone toggle JavaScript functionality on team detail page.
    
        Verifies that:
        - Timezone toggle JavaScript is loaded
        - Toggle button exists and is clickable
        - HTMX is present and functional
        - Button text element is present
        - Individual member overlap times update when timezone is toggled
        """
        team_url = reverse(
            "team_detail",
            kwargs={"session_slug": session.slug, "pk": team.pk},
        )
    
        # Navigate to team detail page (should already be logged in as admin)
        page.goto(team_url)
        page.wait_for_load_state("networkidle")
    
        # Verify the timezone toggle JavaScript function exists
        toggle_function_exists = page.evaluate(
            "typeof window.toggleTimezone === 'function'"
        )
        assert toggle_function_exists, "toggleTimezone function should be defined"
    
        # Verify HTMX is loaded
        htmx_loaded = page.evaluate("typeof htmx !== 'undefined'")
        assert htmx_loaded, "HTMX should be loaded"
    
        # Wait for initial HTMX load to complete - availability section should always be present
        # since all test users have availability set
        page.wait_for_selector(
            "#team-availability-section", state="visible", timeout=5000
        )
    
        # Verify the toggle button exists
        toggle_button = page.get_by_role("button").filter(has_text="Show in")
        expect(toggle_button).to_be_visible()
    
        # Verify the button text element specifically
        button_text_span = page.locator("#toggle-button-text")
        expect(button_text_span).to_be_visible()
    
        # Get initial button text (strip whitespace for comparison)
        initial_button_text = button_text_span.text_content().strip()
        assert len(initial_button_text) > 0, "Button text should not be empty"
    
        # Verify the timezone display span exists
        timezone_span = page.locator("#team-availability-section span.font-bold")
        expect(timezone_span).to_be_visible()
        initial_timezone = timezone_span.text_content().strip()
        assert len(initial_timezone) > 0, "Timezone display should not be empty"
    
        # All test users have the same availability (Monday 9-10 AM UTC),
        # so there should be overlap sections showing
        overlap_times = page.locator(".member-overlap-times")
        assert (
            overlap_times.count() > 0
        ), "Should have overlap sections for team members"
    
        # Emulate browser timezone to Pacific/Kiritimati (UTC+14)
        # This is in the Line Islands, one of the easternmost timezones
        # and extremely unlikely to be anyone's actual timezone
        # This uses Chrome DevTools Protocol to set timezone
        cdp = page.context.new_cdp_session(page)
        cdp.send("Emulation.setTimezoneOverride", {"timezoneId": "Pacific/Kiritimati"})
    
        # Reload the page to pick up the new timezone
        page.reload()
        page.wait_for_load_state("networkidle")
        page.wait_for_selector(
            "#team-availability-section", state="visible", timeout=5000
        )
    
        # Re-get the toggle button and spans after reload
        toggle_button = page.get_by_role("button").filter(has_text="Show in")
        button_text_span = page.locator("#toggle-button-text")
        timezone_span = page.locator("#team-availability-section span.font-bold")
    
        # Get the timezone after reload (should now be Pacific/Kiritimati)
        initial_timezone_local = timezone_span.text_content().strip()
        assert (
            "Pacific/Kiritimati" in initial_timezone_local
        ), f"Expected timezone to be Pacific/Kiritimati, got: {initial_timezone_local}"
    
        # Get initial overlap times in Pacific/Kiritimati timezone
        # 9 AM UTC = 11 PM Pacific/Kiritimati (previous day due to +14 offset)
        overlap_times = page.locator(".member-overlap-times")
        initial_overlap_times_local = overlap_times.first.inner_text()
    
        # Click the toggle button to switch to UTC
        toggle_button.click()
    
        # Wait for any HTMX requests to complete
        page.wait_for_load_state("networkidle")
    
        # Wait for the availability section to be re-rendered after HTMX swap
        page.wait_for_selector("#team-availability-section", state="attached")
    
        # Verify button is still visible after click (ensures page didn't error)
        expect(toggle_button).to_be_visible()
        expect(button_text_span).to_be_visible()
    
        # Get the updated timezone display
        updated_timezone = timezone_span.text_content().strip()
>       assert (
            updated_timezone == "UTC"
        ), f"After toggle, timezone should be UTC, got: {updated_timezone}"
E       AssertionError: After toggle, timezone should be UTC, got: Pacific/Kiritimati
E       assert 'Pacific/Kiritimati' == 'UTC'
E         
E         - UTC
E         + Pacific/Kiritimati

tests/test_complete_session_onboarding_flow.py:1129: AssertionError
```

--- a/tests/test_complete_session_onboarding_flow.py
+++ b/tests/test_complete_session_onboarding_flow.py
@@ -1100,14 +1100,19 @@ class TestCompleteSessionOnboardingFlow:
         button_text_span = page.locator("#toggle-button-text")
         timezone_span = page.locator("#team-availability-section span.font-bold")
 
-        # Get the timezone after reload (should now be Pacific/Kiritimati)
+        # Get the timezone after reload -> get the user's timezone regardless where they are
+        # Detects user's real timezone ("Pacific/Kiritimati" local, "UTC" in CI)
+        # if a user is in UTC, they'll see in their timezone indicator: 'Timezone: UTC'
+        # and the button to toggle will also have the text 'Show in UTC'
         initial_timezone_local = timezone_span.text_content().strip()
+        current_tz_indicator = page.locator("#team-availability-section span.text-sm")
+        indicator_text = current_tz_indicator.text_content().strip()
         assert (
-            "Pacific/Kiritimati" in initial_timezone_local
-        ), f"Expected timezone to be Pacific/Kiritimati, got: {initial_timezone_local}"
+            indicator_text == f"Timezone: {initial_timezone_local}"
+        ), f"Indication should show current user's timezone, but got: {indicator_text}"
 
         # Get initial overlap times in Pacific/Kiritimati timezone
-        # 9 AM UTC = 11 PM Pacific/Kiritimati (previous day due to +14 offset)
+        # For local tests -> 9 AM UTC = 11 PM Pacific/Kiritimati (previous day due to +14 offset)
         overlap_times = page.locator(".member-overlap-times")
         initial_overlap_times_local = overlap_times.first.inner_text()
 

